ophyd/controls/area_detector.py

```diff
@@ -0,0 +1,217 @@
+              value, wait=wait)
+
+    def _make_filename(self, seq=0):
+        uid = str(uuid.uuid4())
+        self._filename = uid
+
+        # The tree is the year / month / day
+        date = datetime.now().date()
+        tree = (str(date.year), str(date.month), str(date.day), '')
+
+        path = os.path.join(self.store_file_path, *tree)
+        self._store_file_path = path
+        self._store_filename = self._file_template.value % (path,
+                                                            self._filename, seq)
+
+        if self.ioc_file_path:
+            path = os.path.join(self.ioc_file_path, *tree)
+            self._ioc_file_path = path
+            self._ioc_filename = self._file_template.value % (path,
+                                                              self._filename,
+                                                              seq)
+
+        else:
+            self._ioc_file_path = self._store_file_path
+            self._ioc_filename = self._store_filename
+
+    def configure(self, *args, **kwargs):
+        super(AreaDetectorFileStore, self).configure(*args, **kwargs)
+        self._uid_cache.clear()
+
+        self._make_filename()
+        self._filestore_res = fs.insert_resource('AD_HDF5',
+                                                 self._store_filename,
+                                                 {'frame_per_point':
+                                                  self._num_images.value})
+
+        # Wait here to make sure we are not still capturing data
+
+        while self._capture.value == 1:
+            print('[!!] Still capturing data .... waiting.')
+            time.sleep(1)
+
+        self._file_path.put(self._ioc_file_path, wait=True)
+        self._file_name.put(self._filename, wait=True)
+        self._write_plugin('AutoIncrement', 1)
+        self._write_plugin('FileNumber', 0)
+        self._write_plugin('FileTemplate', '%s%s_%3.3d.h5', as_string=True)
+        self._write_plugin('NumCapture', 0)
+        self._write_plugin('AutoSave', 1)
+        self._write_plugin('FileWriteMode', 2)
+        self._write_plugin('EnableCallbacks', 1)
+
+        self._capture.put(1, wait=False)
+
+    def _captured_changed(self, value, *args, **kwargs):
+        if value == self._total_images:
+            self._num_captured.clear_sub(self._captured_changed)
+            # Close the capture plugin (closes the file)
+            self._capture.put(0, wait=True)
+
+    def deconfigure(self, *args, **kwargs):
+        self._total_images = self._num_images.value * len(self._uid_cache)
+        self._num_captured.subscribe(self._captured_changed)
+
+        for i, uid in enumerate(self._uid_cache):
+            fs.insert_datum(self._filestore_res, uid, {'point_number': i})
+
+        super(AreaDetectorFileStore, self).deconfigure(*args, **kwargs)
+
+    @property
+    def describe(self):
+        desc = super(AreaDetectorFileStore, self).describe
+
+        size = (self._num_images.value,
```

Comment:
Do you mean that one 2D image should have shape

(1024,1024) and not (1,1024,1024)


On Feb 26, 2015, at 4:06 PM, Thomas A Caswell <notifications@github.com<mailto:notifications@github.com>> wrote:


In ophyd/controls/area_detector.py<https://github.com/NSLS-II/ophyd/pull/71#discussion_r25464254>:

> +            self._capture.put(0, wait=True)
> +
> +    def deconfigure(self, *args, **kwargs):
> +        self._total_images = self._num_images.value * len(self._uid_cache)
> +        self._num_captured.subscribe(self._captured_changed)
> +
> +        for i, uid in enumerate(self._uid_cache):
> +            fs.insert_datum(self._filestore_res, uid, {'point_number': i})
> +
> +        super(AreaDetectorFileStore, self).deconfigure(*args, **kwargs)
> +
> +    @property
> +    def describe(self):
> +        desc = super(AreaDetectorFileStore, self).describe
> +
> +        size = (self._num_images.value,


There is a squeeze is the read-out, this should special case to 2D in the case where self._num_image.value == 1.

—
Reply to this email directly or view it on GitHub<https://github.com/NSLS-II/ophyd/pull/71/files#r25464254>.